src/core/utils.js

```diff
@@ -277,6 +278,40 @@ export function joinAnd(array = [], mapper = item => item, lang = docLang) {
     }
   }
 }
+// STRING HELPERS
+// Takes an array and returns a string that separates each of its items with the proper commas and
+// "and". The second argument is a mapping function that can convert the items before they are
+// joined
+// Finally converts to hyperHTML after joining them
+export function joinAndHyper(
+  array = [],
+  mapper = item => item,
+  lang = docLang
+) {
+  const items = array.map(mapper);
+  if (Intl.ListFormat && typeof Intl.ListFormat === "function") {
+    const formatter = new Intl.ListFormat(lang, {
+      style: "long",
+      type: "conjunction",
+    });
+    return formatter.format(items);
+  }
+  switch (items.length) {
+    case 0:
+    case 1: // "x"
+      return hyperHTML`${items.toString()}`;
+    case 2: // x and y
+      return hyperHTML`${items.join(" and ")}`;
```

Comment:
Here we assume that `items` is an array of `HTMLElement`s, which means that we can't use `.toString()` and `join()` as doing so will generate `[object HTMLElement]`. Instead we should do:

```

    case 1: // "x"
      return items[0];
    case 2: // x and y
      return hyperHTML`${items[0]} and ${items[1]}`;
```

... and similar in `default:`.

BTW, are you already familiar with what hyperHTML does? If not, check out [the documentation](https://viperhtml.js.org/hyperhtml/documentation/) 😁

Reply:
What I was doing was first joining the strings and then , converting them to hyperHTML elements. I was also assuming that the initial array consists of strings. This was where I went wrong :(

Reply:
I will implement as you said but I am unable to get the issue, if the array sent as parameter to joinAnd has raw strings.And then, I convert them to hyperHTML elements at the end.

This way, I will be saved to use hyperHTML on the same text

Reply:
>if the array sent as parameter to joinAnd has raw strings

You forgot to change the return value to use hyperHTML, that's why 😁

Reply:
Can I use hyperHTML.wire(..)..
To join elements in default

Reply:
I don't think that would be any useful there. You can pass element array (not mixing strings) into the string template so that the items can join.

Reply:
![Screenshot from 2019-12-04 09-16-52](https://user-images.githubusercontent.com/44131389/70110992-e27cc080-1676-11ea-92f4-81fb6a943c60.png)
Will this way to join elements be fine

Reply:
No, because `items[i]` must be an `HTMLElement` here and doing this will still cause `[object HTMLElement]`. (If `items[i]` is a string then we don't need this new function.)

Reply:
(Note that `` hyperHTML`${element}` `` and just normal `` `${element}` `` are significant different that the former consumes an element as-is but the latter converts it into a string.)

Reply:
Maybe `` const joinedItems = array.slice(-1).map(item => hyperHTML`${items}, `); `` would be good, as it will contain non-string hyperHTML object that does not stringify any element.

Reply:
items is an array of html elements
If we assume the array in the parameter list of htmlJoinAnd consists of hyperHTML elements, then item[i] is also HTMLElement

Reply:
Ah, the `array` should still contain strings, but we want the mapper to return `HTMLElement` instead of string, and thus we assume `items` to be an array of `HTMLElement`. Sorry for the confusion.

Reply:
I got it now

Reply:
> Maybe `` const joinedItems = array.slice(-1).map(item => hyperHTML`${item}, `); `` would be good, as it will contain non-string hyperHTML object that does not stringify any element.

Although , this was best , is this fine?
![Screenshot from 2019-12-04 10-37-06](https://user-images.githubusercontent.com/44131389/70114327-1ad5cc00-1682-11ea-8c7f-2e55f04f8b60.png)

PS: Thanks for bearing with me so much. I will become better soon

Reply:
Looks much better! But it seems the result will unexpectedly start with a comma because the initial value of `joinedItems` is empty. Also we want to make sure that we get a comma before `"and"` so that it can be `"A, B, and C"` instead of `"A, B and C"`.

Personally would prefer `.map()` over the current for-loop, it somehow matches our code style better 😁